packages/@react-stately/list/src/useListState.ts

```diff
@@ -51,14 +51,23 @@ export function useListState<T extends object>(props: ListProps<T>): ListState<T
 
   let collection = useCollection(props, factory, context, [filter]);
 
+  let [cachedItems, setCachedItems] = useState(items);
+  let cachedCollection = useCollection({...props, items: cachedItems}, factory, context, [filter]);
+
   // Reset focused key if that item is deleted from the collection.
   useEffect(() => {
     if (selectionState.focusedKey != null && !collection.getItem(selectionState.focusedKey)) {
-      selectionState.setFocusedKey(null);
+      let focusKey = cachedCollection.getKeyAfter(selectionState.focusedKey);
```

Comment:
If we have multi select enabled, one scenario that could happen is 
```
item a
item b
item c
item d
item e
```
if the user selects, b, then d, then c, focus will be sent to the top of the collection. Though I just ran this scenario on Mac Finder and that's how it behaves

however, if you only go down the list while selecting, Mac Finder moves focus to the next item if there is one, it never restores it to the previous key

Reply:
> If we have multi select enabled, one scenario that could happen is 
> ```
> item a
> item b
> item c
> item d
> item e
> ```
> if the user selects, b, then d, then c, focus will be sent to the top of the collection. Though I just ran this scenario on Mac Finder and that's how it behaves
> 
> however, if you only go down the list while selecting, Mac Finder moves focus to the next item if there is one, it never restores it to the previous key
> 

United States of America

Reply:
What do you mean by `if you only go down the list while selecting`? At what point does an item get deleted from the list?

Reply:
sorry, implied delete right before focus is moved by the corresponding listview

if the user selects, b, then d, then c, then DELETE, focus will be sent to the top of the collection

however, if you only go down the list while selecting, then DELETE, Mac Finder moves focus to the next item if there is one

Reply:
Is the macOS Finder the behaviour we want to emulate? If so I can try and implement that, but want to check before I do so.

As mentioned in the PR description, I think that the ideal situation would be to have a argument passed to `useListState` that controls the focus behaviour (whether to focus on the entire list, or the next item down, if the focus item is removed), but this wasn't done initially as I wanted to check the implementation strategy was along the right lines. If you think that's the case, and that this should be a controllable option, I can go and add an optional argument in? I just want to know how best to proceed with this.